### DISCUSSION TITLE
Resetting the global PATH to its default value on Ubuntu

### DISCUSSION THREAD

**Discussion by SeSodesa** [fish-shell/fish-shell#9429] [Q&A] [answered]
So I am in a bit of a (not so) funny situation with my global `PATH` variable. I would like to use a newer version of Julia that I just downloaded, but my global `PATH` variable gets loaded with an older version prepended to it, so I can't start the newer version. Yes, adding the bin folder to the `PATH` in the first place was stupid, but here we are.

I would modify the global variable, except I have no idea where this global `PATH` gets loaded from. It is not set in `/etc/fish/config.fish`, nor can the setting be found in `$HOME/.config/fish/config.fish`.

How can I set the global `PATH` to the default value, without modyfying my `config.fish` files? Is it inherited from somewhere? Fish is my default shell, so the only place that could be would be the gnome-shell config, but as far as I know I have not touched that.

**Edit:** it looks like I've run

    fish_add_path  /usr/local/julia/1.8.0/bin/

in the past. How do I undo this setting?

**Comment by faho** [comment, marked as answer]
>How can I set the global PATH to the default value? Is it inherited form somewhere?

When your terminal starts fish, it passes on its environment, and that environment is then imported by fish as global variables.

>Yes, adding the bin folder to the PATH in the first place was stupid, but here we are.

Where did you add it? If you added it to the global $PATH (with something like `set -p PATH /path/to/the/thing`), that wouldn't be *stored* anywhere.

You simply need to figure out how you added it, and then you need to undo it.

If you added it to $fish_user_paths, or indirectly via `fish_add_path`, you need to *remove* it from that. This is a *universal* variable (one that fish stores in its fish_variables file) that is loaded on startup and then fish adds the directories in it to $PATH (if they aren't included already).

**Reply by SeSodesa** [reply]
See the edit in the opening post. It looks like I've at least done a

    fish_add_path  /usr/local/julia/1.8.0/bin/

in the past.

**Reply by SeSodesa** [reply]
Also:

    >> set --show fish_user_paths
    $fish_user_paths: set in universal scope, unexported, with 0 elements

**Reply by SeSodesa** [reply]
The last bit was because I did a

    set -U fish_user_paths

to try and alleviate the current issue.

**Reply by faho** [reply]
Okay, so here's what happens:

1. You've added the path to $fish_user_paths
2. In its startup path, your system starts a fish that then starts your desktop
3. This fish loads $fish_user_paths, adds it to $PATH
4. The processes it starts inherit your old julia path in $PATH
5. Your terminal process now has the old julia path in its $PATH (because its parent had it)
6. Any process your terminal process starts will inherit the old julia path in $PATH

In short: Reboot. You've already removed the path, now you just need to reboot to clean your terminal's environment.

**Reply by SeSodesa** [reply]
Aaand that did it. Forgot to log out. Well played, me.
